packages/eslint-plugin/tests/rules/no-floating-promises.test.ts

```diff
@@ -754,6 +754,53 @@ promise().then(() => {});
         },
       ],
     },
+
+    {
+      code: `
+        declare module 'abc' {
+          export function it(name: string, action: () => void): void;
+        }
+
+        it('...', () => {});
+      `,
+      options: [
+        {
+          allowForKnownSafeCalls: [
+            { from: 'package', name: 'it', package: 'abc' },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
+        declare module 'node:test' {
+          export function it(name: string, action: () => void): void;
+        }
+
+        it('...', () => {});
+      `,
+      options: [
+        {
+          allowForKnownSafeCalls: [
+            { from: 'package', name: 'it', package: 'node:test' },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
+        import { it } from 'node:test';
+
+        it('...', () => {});
+      `,
+      options: [
+        {
+          allowForKnownSafeCalls: [
+            { from: 'package', name: 'it', package: 'node:test' },
+          ],
+        },
+      ],
+    },
```

Comment:
Add test for local modules.

e.g. `declare module './foo'`. It should pass with `{ from: 'file', name: '...' }` but fail with `{ from: 'package', package: './foo' }`.

Reply:
D'oh, I also just realized we already have a dedicated test file for this: `packages/type-utils/tests/TypeOrValueSpecifier.test.ts`. I'll move non-cursory tests there.